lib/common/mock.c

```diff
@@ -285,6 +287,26 @@ __wrap_fopen(const char *pathname, const char *mode)
     }
 }
 
+#ifdef HAVE_FOPEN64
```

Comment:
Alternatively, `__wrap_fopen()` could go in an `#else` CPP block, with this block guarded by `#if defined(HAVE_FOPEN64) && defined(_FILE_OFFSET_BITS) && (_FILE_OFFSET_BITS == 64) && (SIZEOF_LONG < 8)`... which is more verbose, granted.

I haven't looked into the details of the situation, but it seems like we don't actually need `__wrap_fopen()` if we're using `__wrap_fopen64()`. The better readability of the CPP directives might be worth it though.